[NRTM](GH-3088) Add colors and styles from ModernUI for official controls

Evangelink opened the pull request:
**Describe the changes you have made to improve this project**

Provides colors, brushes and styles for the official controls implemented in ModernUI.

Buttons:
![image](https://user-images.githubusercontent.com/11340282/61796400-9c406680-ae25-11e9-923d-6958e5a77d31.png) ![image](https://user-images.githubusercontent.com/11340282/61796405-9f3b5700-ae25-11e9-9926-9eb247ab727f.png)

Checkboxes:
![image](https://user-images.githubusercontent.com/11340282/61796459-bc702580-ae25-11e9-8d89-edccd8934554.png) ![image](https://user-images.githubusercontent.com/11340282/61796461-bed27f80-ae25-11e9-9e10-1422dfb2fdc8.png)

Radiobuttons:
![image](https://user-images.githubusercontent.com/11340282/61796468-c1cd7000-ae25-11e9-8303-893b8d136947.png) ![image](https://user-images.githubusercontent.com/11340282/61796473-c3973380-ae25-11e9-9970-397bc894f877.png)


**Closed Issues**

Fix #3088

Evangelink committed:
MUI colors and brushes

Evangelink committed:
MUI Buttons

Evangelink commented on src/MahApps.Metro/Styles/Themes/GeneratorParameters.json:
```diff
@@ -91,7 +179,94 @@
+        "MahApps.Colors.Mui.SubMenuText": "#99000000",
+        "MahApps.Colors.Mui.SubMenuTextHover": "#66000000",
+        "MahApps.Colors.Mui.SubMenuTextSelected": "#333333",
+        "MahApps.Colors.Mui.WindowHeaderGradient": "<LinearGradientBrush x:Key=\"MahApps.Brushes.Mui.WindowHeaderGradient\" StartPoint=\"0, 0\" EndPoint=\"0, 1\" Opacity=\".1\">\r\n<GradientStop Offset=\"0\" Color=\"{DynamicResource MahApps.Colors.Mui.AccentColor}\" />\r\n<GradientStop Offset=\".3\" Color=\"{DynamicResource MahApps.Colors.Mui.AccentColor}\" />\r\n<GradientStop Offset=\"1\" Color= ...
```
Dark and White theme not only changes the color but uses completely different brushes so I hacked the json generator to generate the full brush. I am open to better suggestions.

Evangelink replied:
Is it ok for you @punker76?

Evangelink commented:
@punker76 Could you confirm this is going the direction you were thinking about?

Evangelink committed:
MUI fonts, sizes and shared content

Evangelink committed:
MUI calendar

Evangelink committed:
MUI CheckBox

Evangelink committed:
MUI ComboBox

Evangelink committed:
MUI ContextMenu

Evangelink committed:
MUI DataGrid

Evangelink committed:
MUI DatePicker

Evangelink committed:
MUI Label

Evangelink committed:
MUI ListBox

Evangelink committed:
MUI ListView

Evangelink committed:
MUI GridSplitter

Evangelink committed:
MUI Hyperlink

Evangelink committed:
MUI TreeView

Evangelink committed:
MUI ToolTip

Evangelink committed:
MUI TextBox

Evangelink committed:
MUI TextBlock

Evangelink committed:
MUI PasswordBox

Evangelink committed:
MUI ProgressBar

Evangelink committed:
MUI RadioButton

Evangelink committed:
MUI ScrollBar

Evangelink committed:
MUI Slider

Evangelink committed:
MUI MenuItem

Evangelink commented on src/MahApps.Metro/Styles/Controls.DatePicker.xaml:
```diff
@@ -307,4 +307,173 @@
+                                <ColumnDefinition Width="*" />
+                                <ColumnDefinition Width="Auto" />
+                            </Grid.ColumnDefinitions>
+                            <!-- TODO: MUI - Replace or implement this control -->
```
TODO

Evangelink replied:
@punker76 I can either use a metro circle button here or import this `ModernButton` from MUI, any preference?

Evangelink commented on src/MahApps.Metro/Styles/Controls.ProgressBar.xaml:
```diff
@@ -123,4 +123,141 @@
+        <Setter Property="Template">
+            <Setter.Value>
+                <ControlTemplate TargetType="ProgressBar">
+                    <!-- TODO: MUI - Import controls or find replacement -->
```
TODO

Evangelink commented on src/MahApps.Metro/Styles/Controls.DataGrid.xaml:
```diff
@@ -623,4 +623,393 @@
 
+    <!-- MUI -->
+
+    <Style x:Key="{ComponentResourceKey TypeInTargetAssembly={x:Type DataGrid}, ResourceId=DataGridSelectAllButtonStyle}" TargetType="{x:Type Button}"
```
Shall we change this key?

Evangelink commented on src/MahApps.Metro/Styles/Controls.ListView.xaml:
```diff
@@ -363,4 +363,181 @@
+        </Setter>
+    </Style>
+
+    <Style x:Key="{x:Static GridView.GridViewStyleKey}" TargetType="{x:Type ListView}">
```
Shall we change this key?

Evangelink commented on src/MahApps.Metro/Styles/Controls.MenuItem.xaml:
```diff
@@ -0,0 +1,224 @@
+    <Geometry x:Key="RightArrow">M 0,0 L 4,3.5 L 0,7 Z</Geometry>
+    <Geometry x:Key="Checkmark">M 15,21L 22,28L 32.9999,14.0001L 35.9999,17.0001L 22,34L 12.0001,24L 15,21 Z</Geometry>
+
+    <Style x:Key="{x:Static MenuItem.SeparatorStyleKey}" TargetType="{x:Type Separator}">
```
Shall we change this key?

Evangelink commented on src/MahApps.Metro/Styles/Controls.MenuItem.xaml:
```diff
@@ -0,0 +1,224 @@
+        </Setter>
+    </Style>
+
+    <ControlTemplate x:Key="{ComponentResourceKey TypeInTargetAssembly={x:Type MenuItem}, ResourceId=SubmenuItemTemplateKey}"
```
Shall we change this key?

Evangelink commented on src/MahApps.Metro/Styles/Controls.Scrollbars.xaml:
```diff
@@ -286,4 +286,304 @@
+        </Setter>
+    </Style>
+
+    <Style x:Key="{x:Static GridView.GridViewScrollViewerStyleKey}" TargetType="{x:Type ScrollViewer}">
```
Shall we change this key?

Evangelink committed:
Update size, font and shared names

Evangelink commented:
This comment is not directly related to this PR but adding even more resources made me wonder if I don't prefer to have the theme after `MahApps` rather than at the end:
`MahApps.Styles.ListBox` -> `MahApps.Metro.Styles.ListBox`
`MahApps.Styles.ListBox.Win10` -> `MahApps.Win10.Styles.ListBox`

and also for font size instead of `MahApps.Sizes.Font.XXX` having `MahApps.Fonts.Size.XXX` and then changing `MahApps.Font.Default` into `MahApps.Font.Family.Default`.

@punker76 WDYT?

Evangelink committed:
Merge new resource dictionaries

Evangelink committed:
Add button, checkbox and radiobutton examples

timunie commented:
> This comment is not directly related to this PR but adding even more resources made me wonder if I don't prefer to have the theme after `MahApps` rather than at the end:
> `MahApps.Styles.ListBox` -> `MahApps.Metro.Styles.ListBox`
> `MahApps.Styles.ListBox.Win10` -> `MahApps.Win10.Styles.ListBox`
> 
> and also for font size instead of `MahApps.Sizes.Font.XXX` having `MahApps.Fonts.Size.XXX` and then changing `MahApps.Font.Default` into `MahApps.Font.Family.Default`.
> 
> @punker76 WDYT?

Hi Guys, 

BTW is it possible to provide a few different ResourceDictionarys where always just one of them will be used in App.xaml merged Dictionary? So one for Metro-Style, one for Win10-Style, one for MUI-Style and so on? I know that this can be manually done, but it would be cool to have it. 

Happy coding 
Tim

Evangelink committed:
MUI Date

punker76 commented:
> @punker76 Could you confirm this is going the direction you were thinking about?

I think so. My first 2 points after first review

- It's maybe better to merge these styles inside it's own resource dictionary `Controls.MUI.xaml` just as @timunie suggested.
- The context menu style for Metro is broken, because it's overridden now with the MUI style

Evangelink committed:
Fix some mui style keys

Evangelink commented:
@punker76 When creating `Controls.MUI.xaml` shall I also set the default style for the components?

timunie commented:
@Evangelink  /cc @punker76 

Here is my suggetion with two examples [https://github.com/timunie/MahApps.Metro/tree/DefaultResourceDictionarys](https://github.com/timunie/MahApps.Metro/tree/DefaultResourceDictionarys) 
I merged all the Controls inside `pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml` and setted my default controls in the new ResourceDictionary. That way the user has still access to all the Styles, if someone want to mix them up.

```
<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
                    xmlns:Controls="clr-namespace:MahApps.Metro.Controls"
                    xmlns:Converters="clr-namespace:MahApps.Metro.Converters">

    <ResourceDictionary.MergedDictionaries>
        <ResourceDictionary Source="pack://application:,,,/MahApps.Metro;component/Styles/Controls.xaml" />
    </ResourceDictionary.MergedDictionaries>

    <!--  note: default style for textblock is now in MetroWindow !!!  -->

    <Style BasedOn="{StaticResource MahApps.Styles.Slider.Win10}" TargetType="Slider" />
    <Style BasedOn="{StaticResource MahApps.Styles.ToggleSwitchButton.Win10}" TargetType="Controls:ToggleSwitchButton" />
    <Style BasedOn="{StaticResource MahApps.Styles.ToggleSwitch.Win10}" TargetType="Controls:ToggleSwitch" />
    <Style BasedOn="{StaticResource MahApps.Styles.RangeSlider.Win10}" TargetType="Controls:RangeSlider" />
</ResourceDictionary>
```

Happy coding
Tim

punker76 committed:
Move MUI controls to Styles/MUI

- Styles/MUI/Controls.xaml contains all styles for MUI
- Styles/Controls.MUI.xaml sets all MUI styles to controls

punker76 committed:
Fix generated MUI brushes and colors

punker76 committed:
Revert MUI samples

punker76 committed:
Rename Scrollbars -> Scrollbar

punker76 committed:
MahApps.Fonts.Default.Mui -> MahApps.Fonts.Family.Default.Mui

punker76 committed:
MahApps.Fonts.Fixed.Mui -> MahApps.Fonts.Family.Fixed.Mui

punker76 committed:
MahApps.Sizes.Font -> MahApps.Font.Size
